src/disputables/index.js

```diff
@@ -0,0 +1,261 @@
+    const appId = await disputableContract.appId()
+
+    if (DISPUTABLE_ACTIONS.has(appId)) {
+      const { abi, entity, fn, scriptPosition } = DISPUTABLE_ACTIONS.get(appId)
+
+      // Get disputed action script
+      const disputableAppContract = getContract(disputableAddress, abi)
+      const result = await disputableAppContract[fn](disputableActionId)
+      const evmScript = result[scriptPosition]
+
+      // Get long and short description corrsponding to the disputed action.
+      const [
+        disputedActionRadspec,
+        disputedActionText,
+      ] = await describeActionScript(evmScript, organization)
+
+      const disputedActionDescription = {
+        disputedActionText,
+        disputedActionRadspec,
+
+        // Build URL where the disputed action is taking place
+        disputedActionURL: buildDisputedActionUrl(
+          organization,
+          disputableAddress,
+          entity,
+          disputableActionId
+        ),
+      }
+
+      // Cache disputed action description
+      cachedDescriptions.set(disputeId, disputedActionDescription)
+
+      return disputedActionDescription
+    }
+  } catch (err) {
+    console.error('Error describing disputable action', err)
+  }
+
+  return { disputedActionRadspec: 'No description' }
+}
+
+/**
+ * Describes the disputed action script using @aragon/connect
+ * @param {String} evmScript The EVM script to describe
+ * @param {String} organization Address of the organization in question
+ * @returns {Array} Array of items where the first item is an array of transaction requests in the path of the disputed action
+ *                    and the second item, the app belonging to the organization where the disputed action is taking place.
+ */
+async function describeActionScript(evmScript, organization) {
+  const org = await connect(organization, 'thegraph', {
+    chainId: env('CHAIN_ID'),
+  })
+  const apps = await org.apps()
+
+  const transactionRequests =
+    (await describeScript(evmScript, apps, org.provider)) || []
+
+  // In order to get the terminal action/s we must search through the `transactionRequests` children (if any)
+  const terminalActions = getTerminalActions(transactionRequests)
+  // In most cases we'll have just a single action to describe
+  // If we do have multiple, we use the first action.
+  const terminalAction = terminalActions[0]
+  const terminalActionAppAddress = terminalAction.to
+
+  // Find the App object corresponding to the app where the terminal action will take place
+  const app = apps.find(app =>
+    addressesEqual(app.address, terminalActionAppAddress)
+  )
+
+  // Get disputed action long description
+  const disputedActionRadspec = terminalActions.length
+    ? buildDisputedActionRadspec(terminalActions)
+    : 'No description'
+
+  // Get disputed action short text
+  // Find method corresponding to the function of the disputed action.
+  const method = await findAppMethodFromIntent(app, terminalAction)
```

Comment:
😅 I didn't know this existed on the connect export list (it doesn't appear to be documented).

I can roughly figure out what it's doing, but I'm not sure it's actually meant to be exported as an API from Connect! Did Gabi mention this was available?

Reply:
haha yes when i was working on this i mentioned @0xGabi what data we needed to have, and he mentioned about this function. Maybe he exposed it so we could use it in this case.